azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -358,7 +362,7 @@ public void submitFlow(final int execId) throws ExecutorManagerException {
 
     final FlowRunner runner =
         new FlowRunner(flow, this.executorLoader, this.projectLoader, this.jobtypeManager,
-            this.azkabanProps);
+            this.azkabanProps, this.azkabanEventReporter);
```

Comment:
Have you considered adding the reporter to the flowRunner instance here instead?
And check if it is null. If it is, don't add it so that the report call won't be called and extra information won't be logged.

Reply:
I do not follow. Are you saying not to initialize `FlowRunner` with the event reporter in-case the reporter is null. Still, I would need to check if it's available or not while handling the event? Or, not to add the flowListener if the event reporter is null? Please let me know.

Reply:
I meant " not to add the flowListener if the event reporter is null". Does it make sense?

Reply:
done.